### ISSUE TITLE
Support WebSockets

### ISSUE THREAD

**Issue by PatrickG** [UnlimitedBytes/adapter-uwebsockets#1]
I think it would be nice, if this adapter would support websockets that can be handled from within a sveltekit app.
Maybe with the new filestructure, we can use `routes/ws/+websocket.js` files.

I guess we would need to create a vite plugin, to support websockets in dev mode.

How a +websocket.js file could look like:
```ts
// routes/example/+websocket.js
export function OPEN({ ws }: { ws: WebSocket }) {
}

export function MESSAGE({ ws, message, isBinary }: { ws: WebSocket; message: ArrayBuffer; isBinary: boolean }) {
}

export function CLOSE({ ws, code, message }: { ws: WebSocket; code: number; message: ArrayBuffer }) {
}
```

OR a default export of the [WebSocketBehavior](https://unetworking.github.io/uWebSockets.js/generated/interfaces/WebSocketBehavior.html) type
```ts
// routes/example/+websocket.js
/** @type import('uWebSocket').WebSocketBehavior */
export default {
  open(ws) {},
  message(ws, message, isBinary) {},
  close(ws, code, message) {},
};
```

But it would be good to have the `app` available in these functions as well, to [publish](https://unetworking.github.io/uWebSockets.js/generated/interfaces/TemplatedApp.html#publish) a message. Or maybe just a `publish` function that is passed as parameter to these functions.

**Comment by UnlimitedBytes** [comment]
Yes, this feature is already planned but before I focus on implementing additional features I wan't to make sure everything already implemented works nice and smooth. This is still in Preview/Alpha stage and although it mostly works there are still some significant bugs/not implemented behavior.

For example I'm currently working on POST/PUT/DELETE/etc. requests as we don't support them right now. It's challenging implementing these kinds of features without dropping the performance gain due to how sveltekit exposes itself to adapters.

Anyhow it's planned and will be included before we hit a release candidate! :)

**Comment by PatrickG** [comment]
Okay. I thought all features are implemented already.
